web/views.py

```
@@ -5291,6 +5292,175 @@
+            else:
+                messages.error(request, "Failed to mint NFT badge. Please try again later.")
+        except ImportError:
+            messages.error(request, "NFT service module not available. Please contact technical support.")
+        except ValueError as e:
+            messages.error(request, f"Invalid data format: {str(e)}")
+        except ConnectionError:
+            messages.error(request, "Failed to connect to blockchain network. Please try again later.")
+        except Exception as e:
+            messages.error(request, f"Error minting NFT badge: {str(e)}")
+            # Log unexpected errors for debugging
+            logger.exception(f"Unexpected error during NFT minting: {achievement_id}, {wallet_address}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)